### PR TITLE
Developers Guide : Object Lifecycle Management

### PR DESCRIPTION
**Pull request by nguptaopensds**

### CODE REVIEW

File: readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+APIs are a set of functions and procedures that allow for the creation of applications that access data and features of other applications, services or operating system. In OpenSDS lifecycle management feature we are using APIs to communicate with different storage backend.
+
+# Write new API
+Before starting to write the APIs, it is required to have basic understanding of XML and JSON syntax.
+
+## How to write the APIs
+
+### There are three major steps involved:
+    a) All APIs goes into the 's3' module of the OpenSDS multi-cloud package, i,e. opensds/multi-cloud/api/pkg/s3 directory
+    b) The corresponding router for the API are defined same directory.
```
Comment (kumarashit) [inline readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md]: > 
```suggestion
    b) The corresponding router for the API are defined **in the** same directory.
```

File: readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+a7288d668842        opensdsio/multi-cloud-api         "/api"                   6 hours ago         Up 6 hours          0.0.0.0:8089->8089/tcp                               multi-cloud_api_1_783889b02933
+247d70f62574        wurstmeister/kafka:2.11-2.0.1     "start-kafka.sh"         12 days ago         Up 6 hours          0.0.0.0:9092->9092/tcp                               multi-cloud_kafka_1_e666cc1a0f3b
+a2c136ae7421        opensdsio/multi-cloud-backend     "/backend"               12 days ago         Up 6 hours                                                               multi-cloud_backend_1_1983cccd8ac9
+eea40a48de97        wurstmeister/zookeeper            "/bin/sh -c '/usr/sb…"   12 days ago         Up 6 hours          22/tcp, 2888/tcp, 3888/tcp, 0.0.0.0:2181->2181/tcp   multi-cloud_zookeeper_1_6643a83ddcdd
+5dbac9cca10e        mongo                             "docker-entrypoint.s…"   12 days ago         Up 6 hours          0.0.0.0:27017->27017/tcp                             multi-cloud_datastore_1_a0281d684ef3
+```
+
+To check if the API changes are reflected in database, install mongoDB compass and check the metadata of the buckets
```
Comment (kumarashit) [inline readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md]: Should we add a note that 'mongoDB compass' is the GUI for mongoDB

File: readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
+# Developer Guide: Object Lifecycle Management
+# Overview
+This guide is intended for Object Lifecycle feature developers. It provides detailed steps to write new API using OpenSDS. To understand the feature and requirements in detail, please refer to the Object Lifecycle Management user guide and Object Lifecycle design doc.
```
Comment (sfzeng) [inline readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md]: It's not used to write new API using OpenSDS, but to do secondary development based on existed lifecycle APIs.

File: readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md (hunk 1 above)
Comment (sfzeng) [reply readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md]: It's not used to create new API, so this chapter, Write new API, is not necessary.
As an alternative, for secondary development, you can provide the examples of creating lifecycle rule, deleting lifecycle rule, disable lifecycle rule, etc.

File: readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+# APIs
+APIs are a set of functions and procedures that allow for the creation of applications that access data and features of other applications, services or operating system. In OpenSDS lifecycle management feature we are using APIs to communicate with different storage backend.
+
+# Write new API
+Before starting to write the APIs, it is required to have basic understanding of XML and JSON syntax.
+
+## How to write the APIs
+
+### There are three major steps involved:
+    a) All APIs goes into the 's3' module of the OpenSDS multi-cloud package, i,e. opensds/multi-cloud/api/pkg/s3 directory
```
Comment (anvithks) [inline readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md]: goes into -> go into

File: readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nguptaopensds) [reply readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md]: Done

File: readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md (hunk 4 above)
Comment (nguptaopensds) [reply readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md]: Done

File: readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md (hunk 2 above)
Comment (nguptaopensds) [reply readthedocs/starterguides/developerguides/object_lifecycle_management_developerguide.md]: Done

**Pull request closed without merging**